client.go

```diff
@@ -273,3 +298,29 @@ func (p httpClientParams) buildClient() (*http.Client, error) {
 	transport := &http.Transport{TLSClientConfig: config}
 	return &http.Client{Transport: transport, Timeout: p.timeout}, nil
 }
+
+func (c *KeywhizHTTPClient) getWithRetry(url string) (resp *http.Response, err error) {
+	t := *c.url
+	t.Path = path.Join(c.url.Path, url)
+
+	b := &backoff.Backoff{
+		//These are the defaults
+		Min:    c.params.minBackoff,
+		Max:    c.params.maxBackoff,
+		Jitter: true,
+	}
+
+	for i := 0; i < c.params.maxRetries; i++ {
+		now := time.Now()
+		resp, err = c.httpClient.Get(t.String())
+		if err != nil || resp.StatusCode == http.StatusOK {
```

Comment:
I'm wondering if there's any other status codes we shouldn't retry for.  404 maybe.

Reply:
probably 401, 403 and generally most (if not all 4xx)? I think we should mainly care about 5xx. I will pull this into `isIntermittent` method